### PR TITLE
internal/olm/client{,.go}: refactor Client to its own package

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** move functions that use `sigs.k8s.io/controller-runtime/pkg/client.Client` directly to `internal/olm/client`.

**Motivation for the change:** future packages that use OLM API's should be able to share a client.

**Note:** this PR mostly refactors code as-is except for `Do{Create,Delete}`.

### CODE REVIEW

File: internal/olm/client.go [hunk 1]
```diff
@@ -249,135 +232,31 @@ func (c Client) doRequest(ctx context.Context, url string) (*http.Response, erro
...
-func decodeResources(rds ...io.Reader) ([]unstructured.Unstructured, error) {
-	var objs []unstructured.Unstructured
+func toObjects(us ...*unstructured.Unstructured) (objs []runtime.Object) {
+	for _, u := range us {
+		objs = append(objs, u)
+	}
+	return objs
+}
+
+func decodeResources(rds ...io.Reader) (objs []*unstructured.Unstructured, err error) {
```
Comment (joelanford) [inline internal/olm/client.go]: Any reason to change this from `[]unstructured.Unstructured` to `[]*unstructured.Unstructured`, which causes all the cascading changes in other function signatures?

If it's for the `toObjects()` function that requires a pointer to satisfy the `runtime.Object` interface, could we update that to:
```go
func toObjects(us ...unstructured.Unstructured) (objs []runtime.Object) {
	for _, u := range us {
		objs = append(objs, &u)
	}
	return objs
}
```

File: internal/olm/client.go [hunk 2]
```diff
@@ -386,7 +265,7 @@ func filterResources(resources []unstructured.Unstructured, filter func(unstruct
 	return filtered
 }
 
-func (c Client) getSubscriptionCSV(ctx context.Context, subKey types.NamespacedName) (*types.NamespacedName, error) {
+func (c Client) getSubscriptionCSV(ctx context.Context, subKey types.NamespacedName) (types.NamespacedName, error) {
 	var csvKey *types.NamespacedName
```
Comment (joelanford) [inline internal/olm/client.go]: If we're returning a non-pointer, we might as well change this to:

```suggestion
	var csvKey types.NamespacedName
```

And then update line 280 to use a non-pointer when it sets `csvKey`

File: internal/olm/client.go (hunk 1 above)
Comment (estroz) [reply internal/olm/client.go]: I don't mind changing pointers back to values; this change is part of an original draft that I ended up not going with.

**Pull request merged**